app/assets/javascripts/quby/answers/validation.js

```diff
@@ -189,6 +208,48 @@
     return !failed;
   };
 
+  function parsePartialDate(inputs) {
+    var value = function(placeholder) {
+      var val = inputs.filter("[placeholder=" + placeholder + "]").first().val();
+      if(val === undefined || val == "")
+        return null;
+      if(intVal = parseInt(val)) {
+        return intVal;
+      } else {
+        throw "invalidDate";
+      }
+    };
+
+    var year   = value('YYYY') || 2000;
```

Comment:
Misschien een idee om deze functie op te delen in een functie die de waardes uit de inputs haalt (waarbij bv input niet aanwezig een null is, een lege input '' en anders de integer) gevolgd door een functie die de defaults erin stopt (alleen voor nulls) en valideert? Of zelfs 3 misschien.